internal/providers/terraform/aws/ssm_activation.go

```diff
@@ -0,0 +1,63 @@
+package aws
+
+import (
+	"github.com/infracost/infracost/internal/schema"
+	"github.com/shopspring/decimal"
+)
+
+func GetSSMActivationRegistryItem() *schema.RegistryItem {
+	return &schema.RegistryItem{
+		Name:  "aws_ssm_activation",
+		RFunc: NewSSMActivation,
+	}
+}
+
+func NewSSMActivation(d *schema.ResourceData, u *schema.UsageData) *schema.Resource {
+	region := d.Get("region").String()
+
+	tier := "Standard"
+	instanceCount := decimal.Zero
```

Comment:
How about this for the logic:
1. Don't show standard cost component as it's free.
2. If `registration_limit` in Terraform is set and it's > 1000 and `aws_ssm_activation.managed_instances` is not set, show the advanced price but no costs (i.e. quantity is nil as we don't know).
3. If `aws_ssm_activation.managed_instances` is set in infracost-usage.yml and it's > 1000, show advanced pricing with the value as quantity.

@aliscott any thoughts/ideas on if we should validate values? e.g. if the registration_limit is less than aws_ssm_activation.managed_instances, should we show an error to say it can't be more than the limit? if so, how should the code handle that? throw an exception seems too harsh?

Reply:
So with this the registration_limit can be below 100 if you want to take advantage of the Advanced tier features. Advanced tier isn't just for when you have more than 1000 instances. You need to switch to advanced tier if you want  the additional features i.e. Session manager or Windows application patching. This flag is set at the account/region within AWS. There isn't a TF resource so you don't know whether it's set or not. I think I've added logic to cover the scenarios.